File: rhdetail_index/src/main/scala/cn/bigdata/rhdetailindex/utils/DBUtils.scala
package cn.bigdata.rhdetailindex.utils

import java.sql.{Connection, DriverManager}

import com.typesafe.config.{Config, ConfigFactory}

object DBUtils {
  Class.forName("com.mysql.jdbc.Driver")
  def getConnection: Connection = {
    val config: Config = ConfigFactory.load()
    DriverManager.getConnection(config.getString("MYSQL_DRIVER_URL"), config.getString("MYSQL_USERNAME"), config.getString("MYSQL_PWD"))
  }
  def close(conn: Connection): Unit = {
    try{
      if(!conn.isClosed() || conn != null){
        conn.close()
      }
    }
    catch {
      case ex: Exception => {
        ex.printStackTrace()
      }
    }finally {
      conn.close()
    }
  }
}

File: code/src/main/java/com/jinke/tianyan/util/DBUtils.java
package com.jinke.tianyan.util;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DBUtils {
    private static Connection con;
    //在静态代码块中创建与数据库的连接
/*    static{
        try{
            String driver = "com.mysql.jdbc.Driver";
            Class.forName(driver);
            con = DriverManager.getConnection(Global.MYSQL_DRIVER_URL, Global.MYSQL_USERNAME, Global.MYSQL_PWD);
        }catch(Exception e){
            throw new RuntimeException(e.getMessage(),e);
        }
    }*/

    public static Connection getCon(){
        try{
            String driver = "com.mysql.jdbc.Driver";
            Class.forName(driver);
            con = DriverManager.getConnection(Global.MYSQL_DRIVER_URL, Global.MYSQL_USERNAME, Global.MYSQL_PWD);
            return con;
        }catch(Exception e){
            throw new RuntimeException(e.getMessage(),e);
        }

    }

    public static void close(Connection conn){
        try{
            if(conn != null){
                try {
                    conn.close();
                } catch (SQLException e) {
                    e.printStackTrace();
                }
            }
        }catch(Exception e){
            e.printStackTrace();
        }

    }
}
